From 51a9f15b10357a4ad1d8b2e50ec14031f68c0d48 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@digia.com>
Date: Fri, 12 Apr 2013 09:41:41 +0300
Subject: [PATCH] support multitouch screens

---
 src/udev-builtin-input_id.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/udev-builtin-input_id.c b/src/udev-builtin-input_id.c
index a062ef7..0c5358e 100644
--- a/src/udev-builtin-input_id.c
+++ b/src/udev-builtin-input_id.c
@@ -108,8 +108,9 @@ static void test_pointers (struct udev_device *dev,
                 return;
         }
 
-        if (test_bit (EV_ABS, bitmask_ev) &&
-            test_bit (ABS_X, bitmask_abs) && test_bit (ABS_Y, bitmask_abs)) {
+        if (test_bit (EV_ABS, bitmask_ev) && (
+            (test_bit (ABS_X, bitmask_abs) && test_bit (ABS_Y, bitmask_abs)) ||
+            (test_bit (ABS_MT_POSITION_X, bitmask_abs) && test_bit (ABS_MT_POSITION_Y, bitmask_abs)))) {
                 if (test_bit (BTN_STYLUS, bitmask_key) || test_bit (BTN_TOOL_PEN, bitmask_key))
                         udev_builtin_add_property(dev, test, "ID_INPUT_TABLET", "1");
                 else if (test_bit (BTN_TOOL_FINGER, bitmask_key) && !test_bit (BTN_TOOL_PEN, bitmask_key))
-- 
1.7.10.4

