From 4fe2e7b7b9a83f36b409892e854aed68dbd45a0a Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@digia.com>
Date: Thu, 23 May 2013 14:37:10 +0300
Subject: [PATCH] beagle: change default resolution and increase VRAM for fb0

---
 include/configs/omap3_beagle.h |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/configs/omap3_beagle.h b/include/configs/omap3_beagle.h
index 1cc46dc..04862d1 100644
--- a/include/configs/omap3_beagle.h
+++ b/include/configs/omap3_beagle.h
@@ -224,7 +224,7 @@
 	"optargs=\0" \
 	"camera=none\0" \
 	"vram=12M\0" \
-	"dvimode=640x480MR-16@60\0" \
+	"dvimode=800x600MR-16@60\0" \
 	"defaultdisplay=dvi\0" \
 	"mmcdev=0\0" \
 	"mmcroot=/dev/mmcblk0p2 ro\0" \
@@ -240,8 +240,10 @@
 		"buddy2=${buddy2} "\
 		"camera=${camera} "\
 		"vram=${vram} " \
+		"omapfb.vram=0:${vram} " \
 		"omapfb.mode=dvi:${dvimode} " \
 		"omapdss.def_disp=${defaultdisplay} " \
+		"consoleblank=0 " \
 		"root=${mmcroot} " \
 		"rootfstype=${mmcrootfstype}\0" \
 	"nandargs=setenv bootargs console=${console} " \
@@ -250,8 +252,10 @@
 		"buddy=${buddy} "\
 		"camera=${camera} "\
 		"vram=${vram} " \
+		"omapfb.vram=0:${vram} " \
 		"omapfb.mode=dvi:${dvimode} " \
 		"omapdss.def_disp=${defaultdisplay} " \
+		"consoleblank=0 " \
 		"root=${nandroot} " \
 		"rootfstype=${nandrootfstype}\0" \
 	"bootenv=uEnv.txt\0" \
@@ -263,8 +267,10 @@
 		"mpurate=${mpurate} " \
 		"buddy=${buddy} "\
 		"vram=${vram} " \
+		"omapfb.vram=0:${vram} " \
 		"omapfb.mode=dvi:${dvimode} " \
 		"omapdss.def_disp=${defaultdisplay} " \
+		"consoleblank=0 " \
 		"root=${ramroot} " \
 		"rootfstype=${ramrootfstype}\0" \
 	"loadramdisk=fatload mmc ${mmcdev} ${rdaddr} ramdisk.gz\0" \
-- 
1.7.10.4

